crates/oxpinyin-python/tests_py/conftest.py

```
@@ -2,13 +2,53 @@
+
+def resolve_input(case: dict) -> str:
+    """Resolves a case's effective input: `input`, or `repeat.unit` × times."""
+    if "input" in case:
+        return case["input"]
+    repeat = case.get("repeat") or {}
+    unit, times = repeat.get("unit", ""), int(repeat.get("times", 0))
+    total = len(unit) * times
+    if total > INPUT_CAP:
+        raise ValueError(
+            f"repeated input is {total} bytes, past the {INPUT_CAP}-byte corpus cap"
+        )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)